ecosystem/sep-0009.md

```diff
@@ -148,8 +148,34 @@ single multi-line string in the `address` field. This allows any address in the
 number of fields. If address parsing is necessary, parsing will be easier since the country, city, and postal code are
 already separate fields.
 
+## Card fields
+
+To pass card (such as credit card) details to the application, the client should pass card details via an object defined
+below. Note, that it's possible to either pass card details to the application, or the token, representing the card.
+This token may be fetched from a third-party source prior, for example, it can be a
+[Stripe token](https://stripe.com/docs/api/tokens/create_card), or any other service offering similar functionality may
+be used. When token is used, application should notify clients about the type of the token that it is expecting to
+receive. Usually, application would require either `token`, or set of: `number`, `expiration_date`, `cvc` and
+`holder_name` to be provided, but some applications may require extra fields.
+
+| Name                     | Type   | Description                                                                                 |
+|--------------------------| ------ | ------------------------------------------------------------------------------------------- |
+| `card.number`            | number | Card number                                                                                 |
+| `card.expiration_date`   | date   | Expiration month and year (e.g. `11-29`, November 2029)                                     |
+| `card.cvc`               | number | CVC number (3 digits on the back of the card)                                               |
+| `card.holder_name`       | string | Name of the card holder                                                                     |
+| `card.type`              | string | Type of the bank card used (e.g. debit, credit)                                             |
+| `card.brand`             | string | Brand of the card (e.g. Visa, Mastercard)                                                   |
+| `card.postal_code`       | string | Post or other code used by the card for the billing purposes                                |
+| `card.country_code`      | string | Country code used by the card for the billing purposes                                      |
+| `card.state_or_province` | string | Name of state/province/region/prefecture                                                    |
+| `card.city`              | string | Name of city/town                                                                           |
+| `card.address`           | string | Entire address (country, state, postal code, street address, etc...) as a multi-line string |
```

Comment:
Should we break this into individual fields since we already have `card.country_code`, and `card.postal_code`?

Reply:
I don't think so because addresses can consist of a large variety of parts depending on the region. If anything,  should have two fields for the address (part 1 and part 2?) for things like unit or apartment numbers?